src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -117,6 +122,31 @@ internal class WorkloadInformationProvider : IWorkloadInformationProvider
             }
         }
 
+        public List<int> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
+            var portsToIgnore = new List<int>();
+
+            List<V1Service> portsList = serviceNameSpace.Items.Where(item => item.Metadata != null && 
+            item.Metadata.Annotations != null &&
+            item.Metadata.Annotations.ContainsKey(ServiceAnnotations)).ToList();
+
+            portsList.ForEach(port => {
+                if (port.Metadata.Annotations.TryGetValue(ServiceAnnotations, out string ports)) {
+                    if (ports.Length > 0) {
+                        try {
+                            // ports e.g. "460, 543"
+                            var ignorePorts = ports.Split(",").Select(p => int.Parse(p));
+                            portsToIgnore.AddRange(ignorePorts);
+                        } 
+                        catch {
+                            throw new UserVisibleException(this._operationContext, $"bridgetokubernetes/ignore-ports configuration value {port} is invalid. It should be a comma separated list of integer ports");

```

Comment:
typo, i think this needd to be ports. Once you add UT for bad formatted string it should catch it :)

Reply:
I believe once you add UT, you will see that usage of UserVisableException is wrong, it required some pre-defined format. I think it may need to follow format similar to line 102 ("throw new UserVisibleException(_operationContext, Resources.PortRequiredFormat, externalEndpoint.Name);")